keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,972 @@
+// for extended resource request.
+type DeviceExtendedResourceRequestAllocationResult struct {
+	// ExtendedResourceName is the extended resource name the devices are
+	// allocated for.
+	//
+	// +required
+	ExtendedResourceName string `json:"extendedResourceName" protobuf:"bytes,1,name=extendedResourceName"`
+
+	// Driver specifies the name of the DRA driver whose kubelet
+	// plugin should be invoked to process the allocation once the claim is
+	// needed on a node.
+	//
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	//
+	// +required
+	Driver string `json:"driver" protobuf:"bytes,2,name=driver"`
+
+	// This name together with the driver name and the device name field
+	// identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+	//
+	// Must not be longer than 253 characters and may contain one or more
+	// DNS sub-domains separated by slashes.
+	//
+	// +required
+	Pool string `json:"pool" protobuf:"bytes,3,name=pool"`
+
+	// Device references one device instance via its name in the driver's
+	// resource pool. It must be a DNS label.
+	//
+	// +required
+	Device string `json:"device" protobuf:"bytes,4,name=device"`
+
+	// AdminAccess indicates that this device was allocated for
+	// administrative access. See the corresponding request field
+	// for a definition of mode.
+	//
+	// This is an alpha field and requires enabling the DRAAdminAccess
+	// feature gate. Admin access is disabled if this field is unset or
+	// set to false, otherwise it is enabled.
+	//
+	// +optional
+	// +featureGate=DRAAdminAccess
+	AdminAccess *bool `json:"adminAccess" protobuf:"bytes,5,name=adminAccess"`
+}
+```
+
+A special resource claim object is created to keep track of device allocations for
+extended resource. The resource claim object has following properties:
+
+  * It is namespace scoped, like other resource claim objects.
+  * It is owned by a pod, like other resource claim objects.
+  * It has null `spec`.
+  * Its `status.allocation.devices` and `status.allocation.reservedFor` are
+    used.
+  * It does not have annotation `resource.kubernetes.io/pod-claim-name:` as
+    it is created for the extended resource request in a pod spec, not for a
+    claim in the pod spec.
+
+The special resource claim object lifecycle is managed by the scheduler and
+garbage collector.
+
+  * It is *created* in a namespace when there is a pod with extended resource
+  request, and the extended resource is advertised by `ResourceSlice` and
+  scheduler has fit the pod to a node with the `ResourceSlice`.
+  * It is *created* by the scheduler dynamic extended resource plugin during
+    preBind phase. The in-memory one in the assumed cache is created earlier
+    during reserve phase.
+  * It is *deleted* together with the owning pod's deletion.
+  * It is *deleted* by the scheduler dynamic extended resource plugin during
```

Comment:
Why can't we use ownerRef and k8s gc instead?

Reply:
That's the previous bullet point. This one is about the pod not proceeding to be bound: it doesn't get deleted, so the scheduler has to roll back.

Reply:
In other words, we need to do both depending on the circumstances.

@yliaog: having two sub-bullets under "It is *deleted* ..." might make it clearer that these two alternative outcomes.